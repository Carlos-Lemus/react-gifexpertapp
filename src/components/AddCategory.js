import React, { useState } from 'react'
import PropTypes from "prop-types";

export const AddCategory = ( { setCategories} ) => {

    const [inputValue, setInputValue] = useState("");

    const submitHandle = event => {
        event.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
    
            setInputValue("");
        }
    

        console.log("submitHandle llamado");
    };

    const changeInputHandle = event => {
        setInputValue(event.target.value);

        console.log("changeInputHandle llamado");
    };

    return (
        <form onSubmit={ submitHandle }>
            <p> {inputValue} </p>
            <input type="text" 
                value={ inputValue } 
                onChange={ changeInputHandle } 
                />
        </form>
    );

};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};


