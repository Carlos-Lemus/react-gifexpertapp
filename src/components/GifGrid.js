import React, { useEffect, useState } from 'react';
import PropTypes from "prop-types";
import { useFecthGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from './GifGridItem';

const GifGrid = ( { category } ) => {

    const {data:gifs, loading} = useFecthGifs(category);
    

    return (
        <>
            <h3>{ category }</h3>
            {loading && <p className="animate__animated animate__flash animate__delay-2">Loading...</p>}

            { <ol className="card-grid">
                {
                    gifs.map( image => (
                        <GifGridItem key={image.id} {...image} />
                    ))
                }
            </ol> }

        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};

export default GifGrid;