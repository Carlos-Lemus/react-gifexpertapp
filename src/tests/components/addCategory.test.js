import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import "@testing-library/jest-dom";

describe("Probando el componente <addCategory />", () => {
   
    const setCateogires = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCateogires } /> );
    
    beforeEach(() => {
        jest.clearAllMocks();
       
        wrapper = shallow( <AddCategory setCategories={ setCateogires } /> );
    });

    test("Tomando un snapshot del componente <addCategory />", () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Probando el evento onChange en el input', () => {
        
        const input = wrapper.find("input");
        const value = "Prueba input";
        
        input.simulate("change", {
            target: {
                value: value
            }
        });

        expect( wrapper.find("p").text().trim() ).toBe(value);

    })
    
    test('no debe ser llamada la funcion setCategories', () => {
        
        wrapper.find("form").simulate("submit", {
            preventDefault() {}
        });
        
        expect(setCateogires ).not.toHaveBeenCalled();
    })
    
    test('Debe de llarmarse setCategories', () => {
        
        const value = "One punch Man";

        const input = wrapper.find("input");

        input.simulate("change", {
            target: {
                value: value
            }
        });

        wrapper.find("form").simulate("submit", {
            preventDefault() {}
        });

        expect( setCateogires ).toHaveBeenCalled();

        expect( input.prop("value") ).toBe("");
        console.log(input.text());
    })
    

});