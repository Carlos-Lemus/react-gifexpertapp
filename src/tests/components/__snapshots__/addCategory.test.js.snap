// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <addCategory /> Tomando un snapshot del componente <addCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
